lib/collection/src/shards/local_shard/query.rs

```diff
@@ -196,32 +198,33 @@ impl LocalShard {
     }
 
     /// Rescore list of scored points
+    #[allow(clippy::too_many_arguments)]
     async fn rescore<'a>(
         &self,
         sources: impl Iterator<Item = Cow<'a, Vec<ScoredPoint>>>,
         rescore_query: ScoringQuery,
+        filter: Option<Filter>,
         limit: usize,
+        score_threshold: Option<f32>,
         search_runtime_handle: &Handle,
         timeout: Option<Duration>,
     ) -> CollectionResult<Vec<ScoredPoint>> {
         match rescore_query {
             ScoringQuery::Fusion(Fusion::Rrf) => {
-                let top_rrf = rrf_scoring(sources.map(Cow::into_owned), limit, 0);
+                let mut top_rrf = rrf_scoring(sources.map(Cow::into_owned), limit, 0);
+
+                if let Some(score_threshold) = score_threshold {
```

Comment:
we should use filter in fusion path as well

Reply:
Handled. We will remove it once we merge and propagate all filters to leaf queries, though